### PR TITLE
Adds kafka consumer

### PR DESCRIPTION
**Pull request by djperrefort**
This PR replaces the ``alert_aquisition`` and ``data_upload`` modules with the ``consumer`` module. Some edits are made to the docs, but they need a more dedicated pass through that is beyond this scope.

### CODE REVIEW

File: broker/consumer.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    # 'sasl.kerberos.keytab':
+}
+
+
+class GCSKafkaConsumer(Consumer):
+    """Ingests data from a kafka stream into big_query"""
+
+    def __init__(self, kafka_config, bucket_name, kafka_topic,
+                 pubsub_topic, debug=False):
+        """Ingests data from a kafka stream into big_query
```
Comment (wmwv) [inline broker/consumer.py]: "And stores a copy in GCS"

File: broker/consumer.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+   config = DEFAULT_ZTF_CONFIG.copy()
+   config['sasl.kerberos.keytab'] = '<Path to authentication file>'
+   config['sasl.kerberos.principal'] = '<>'
+   print(config)
+
+   # Create a consumer
+   c = GCSKafkaConsumer(
+       kafka_config=config,
+       bucket_name='my-GCS-bucket-name',
+       kafka_topics='ztf_20200301_programid1',
```
Comment (wmwv) [inline broker/consumer.py]: `kafka_topic`  no "s"

File: broker/consumer.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+       pubsub_topic='my-GCS-PubSub-name',
+       debug=True  # Use debug to run without updating your kafka offset
+   )
+
+   # Ingest alerts one at a time indefinitely
+   c.run()
```
Comment (wmwv) [inline broker/consumer.py]: How do I check the pub sub topic to see these messages?

File: docs/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+sphinx
+sphinx_rtd_theme
```
[the remark this answers is not in the document]
Comment (djperrefort) [reply docs/requirements.txt]: This is the extra requirements file for the docs. I'll add it to the package requirements.

File: broker/consumer.py (hunk 3 above)
Comment (djperrefort) [reply broker/consumer.py]: In GCS select `Pub/Sub` -> `Topics` -> `testing-topic`. Programatically I'm not sure but @cnmazz might know.

**Comment by djperrefort** [comment]
I'll merge once @troyraen has a chance to review.

File: broker/consumer.py [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+   config['sasl.kerberos.principal'] = '<>'
+   print(config)
+
+   # Create a consumer
+   c = GCSKafkaConsumer(
+       kafka_config=config,
+       bucket_name='my-GCS-bucket-name',
+       kafka_topic='ztf_20200301_programid1',
+       pubsub_topic='my-GCS-PubSub-name',
+       debug=True  # Use debug to run without updating your kafka offset
```
Comment (troyraen) [inline broker/consumer.py]: Naively following these instructions and did not create the bucket first. Got an error on this line `debug=True`:

`BadRequest: 400 GET https://www.googleapis.com/storage/v1/b/my-GCS-bucket-name-Troys-test?projection=noAcl: Invalid bucket name: 'my-GCS-bucket-name-Troys-test'`

Then it kept trying to connect, and returning errors. `ctrl-c` had no effect and I had to shut down my terminal.

I suggest we either add the line of code that creates the bucket (not sure what this is) or just a note saying the bucket needs to exist.

File: broker/consumer.py [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+   # Define connection configuration using default values as a starting point
+   config = DEFAULT_ZTF_CONFIG.copy()
+   config['sasl.kerberos.keytab'] = '<Path to authentication file>'
+   config['sasl.kerberos.principal'] = '<>'
+   print(config)
+
+   # Create a consumer
+   c = GCSKafkaConsumer(
+       kafka_config=config,
+       bucket_name='my-GCS-bucket-name',
```
Comment (troyraen) [inline broker/consumer.py]: Bucket name cannot include capital letters

File: broker/consumer.py (hunk 5 above)
Comment (troyraen) [reply broker/consumer.py]: The problem forcing me to shut down the terminal was actually caused by not setting `os.environ['KRB5_CONFIG'] = './krb5.conf'` as detailed in the authentication notebook. This module probably assumes one is already authenticated, and in that case the error returned about the bucket not existing is self-explanatory and not fatal, so the code is probably fine the way it is.

File: broker/consumer.py (hunk 6 above)
Comment (djperrefort) [reply broker/consumer.py]: Fixed - Thanks @troyraen!

**Pull request merged**